frontend/src/app/organization/organization-create/organization-create.component.ts

```diff
@@ -0,0 +1,88 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+import { filter, map, switchMap, take, tap } from 'rxjs/operators';
+import { Subscription } from 'rxjs/Subscription';
+
+import { AuthService } from '../../auth.service';
+import { User } from '../../user';
+import { UserService } from '../../user.service';
+import { Organization } from '../organization.model';
+import { OrganizationService } from '../organization.service';
+
+@Component({
+  selector: 'volontulo-organization-create',
+  templateUrl: './organization-create.component.html',
+  styleUrls: ['./organization-create.component.scss']
+})
+export class OrganizationCreateComponent implements OnInit, OnDestroy {
+  createForm: FormGroup;
+  id: number;
+  inEditMode = false;
+  message = '';
+  createSubscription: Subscription;
+  organization$: Observable<Organization>;
+  user: User;
+
+  constructor(
+    private authService: AuthService,
+    private activatedRoute: ActivatedRoute,
+    private fb: FormBuilder,
+    private organizationService: OrganizationService,
+    private router: Router,
+    private userService: UserService,
+    ) {}
+
+  ngOnInit() {
+    this.authService.user$.pipe(take(1))
+      .subscribe(
+      (user: User) => { this.user = user }
+    );
+
+    this.createForm = this.fb.group({
+       'name': this.fb.control(null, Validators.required),
+       'address': this.fb.control(null, Validators.required),
+       'description': this.fb.control(null, Validators.required),
+      });
+
+    this.organization$ = this.activatedRoute.params.pipe(
+      map(params => params.organizationId),
```

Comment:
The change you made in OrganizationService on lines 34-37 allowed you to chain those observables here, which looks good on the frist glance. However, it caused the side effect in organizations.component on line 38, which forced you to use subscribe() there.

To put more light on this - the change you made broke the pattern that required to subscribe to events emitted from the service. In OrganizationsService, the methods were not directly exposing observables by design - they only emitted events to Subjects instead. This way the asynchronous method call could be decoupled from the subscribers observing the response.

When I found that, I tried once again to find another solution to chain those methods in a clean way and in the same time adhere to the pattern provided in OrganizationService, but I failed. I didn't find another solution except for the one provided by me earlier, where we use two independent subscriptions. 

It doesn't mean that there might not be any better solution, but I can't help more with that. Do you think there is anyone on the team experienced enough in RxJS that could help you solve this in another way? If yes, please ask this person for help. 

The previous solution was as far as I could go with this. The one you have now looks nice here, but it breaks the pattern estabilished in the OrganizationService, which I personally think we shouldn't do.